src/textual/_on.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import Callable, TypeVar
+
+from .css.parse import parse_selectors
+from .css.tokenizer import TokenError
+from .message import Message
+
+DecoratedType = TypeVar("DecoratedType")
+
+
+class OnDecoratorError(Exception):
+    """The selector in the 'on' decorator failed to parse."""
```

Comment:
This error is also used when other things fail, so maybe the docstring should be a bit more generic?